lib/src/components/DateTimePicker/index.tsx

```diff
@@ -39,17 +42,17 @@ export const DateTimePicker = forwardRef<'input', DateTimePickerProps>(
     const formatDate: (date: DateTimePickerProps['value']) => ReturnType<typeof getDate> = date =>
       getDate(date, timeIntervals)
 
-    const [date, setDate] = useState(formatDate(value))
+    const [date, setDate] = useState<Date | null>(formatDate(value))
 
     const handleChange: DateTimePickerProps['onChange'] = newDate => {
       setDate(newDate || null)
-      onChange && onChange(newDate && new Date(newDate))
+      onChange?.(newDate && new Date(newDate))
```

Comment:
I would go with a if here, same as line 54 (and I would keep it consistent across all files)